dotCMS/src/main/java/com/dotcms/metrics/MetricsService.java

```diff
@@ -0,0 +1,282 @@
+            } catch (Exception e) {
+                Logger.error(this, "Failed to register database metrics: " + e.getMessage(), e);
+            }
+        }
+        
+        // Cache Metrics
+        if (MetricsConfig.CACHE_METRICS_ENABLED) {
+            try {
+                new CacheMetrics().bindTo(globalRegistry);
+                Logger.debug(this, "Cache metrics registered");
+            } catch (Exception e) {
+                Logger.error(this, "Failed to register cache metrics: " + e.getMessage(), e);
+            }
+        }
+
+        
+        // HTTP Request Metrics
+        if (MetricsConfig.HTTP_METRICS_ENABLED) {
+            try {
+                new HttpRequestMetrics().bindTo(globalRegistry);
+                Logger.debug(this, "HTTP request metrics registered");
+            } catch (Exception e) {
+                Logger.error(this, "Failed to register HTTP request metrics: " + e.getMessage(), e);
+            }
+        }
+
+        
+        // Tomcat Metrics
+        if (MetricsConfig.TOMCAT_METRICS_ENABLED) {
+            try {
+                new TomcatMetrics().bindTo(globalRegistry);
+                Logger.debug(this, "Tomcat metrics registered");
+            } catch (Exception e) {
+                Logger.error(this, "Failed to register Tomcat metrics: " + e.getMessage(), e);
+            }
+        }
+    }
+    
+    
+    /**
+     * Get the Prometheus meter registry.
+     * 
+     * @return PrometheusMeterRegistry instance, or null if not configured
+     */
+    public PrometheusMeterRegistry getPrometheusRegistry() {
+        return prometheusRegistry;
+    }
+    
+    /**
+     * Get the JMX meter registry.
+     * 
+     * @return JmxMeterRegistry instance, or null if not configured
+     */
+    public JmxMeterRegistry getJmxRegistry() {
+        return jmxRegistry;
+    }
+    
+    /**
+     * Get the global meter registry.
+     * 
+     * @return Global MeterRegistry instance
+     */
+    public MeterRegistry getGlobalRegistry() {
+        return Metrics.globalRegistry;
+    }
+    
+    /**
+     * Get the metrics tagging service for creating tagged metrics.
+     * 
+     * @return MetricsTaggingService instance
+     */
+    public MetricsTaggingService getTaggingService() {
+        return taggingService;
+    }
+    
+    /**
+     * Check if metrics are enabled and initialized.
+     * 
+     * @return true if metrics are enabled, false otherwise
+     */
+    public boolean isEnabled() {
+        return MetricsConfig.ENABLED && (prometheusRegistry != null || jmxRegistry != null);
+    }
+    
+    /**
+     * Clean up resources when the service is destroyed.
+     * Called automatically by CDI during application shutdown.
+     */
+    @PreDestroy
+    public void destroy() {
+        if (!MetricsConfig.ENABLED) {
+            return;
+        }
+        
+        Logger.info(this, "Shutting down Metrics Service");
+        
+        for (MeterRegistry registry : registries) {
```

Comment:
if MeterRegistry is a AutoCloseable i can use this util: CloseUtils

Reply:
MeterRegistry is not AutoCloseable it's an abstract class with a regular close() method. CloseUtils.closeQuietly() only works with
AutoCloseable objects. The current manual approach with try-catch blocks and logging is the correct pattern for this case.